ocs_ci/deployment/baremetal.py

```diff
@@ -22,7 +36,461 @@ class BAREMETALUPI(Deployment):
     def __init__(self):
         logger.info("BAREMETAL UPI")
         super().__init__()
-        # TODO: OCP,deployment
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super().__init__()
+            self.helper_node_details = load_auth_config()['baremetal']
+            self.mgmt_details = load_auth_config()['ipmi']
+
+        def deploy_prereq(self):
+            """
+             Pre-Requisites for Bare Metal UPI Deployment
+            """
+            super(BAREMETALUPI.OCPDeployment, self).deploy_prereq()
+            # create manifest
+            self.create_manifest()
+            # create ignitions
+            self.create_ignitions()
+            self.kubeconfig = os.path.join(self.cluster_path, config.RUN.get('kubeconfig_location'))
+            bootstrap_path = os.path.join(config.ENV_DATA.get('cluster_path'), constants.BOOTSTRAP_IGN)
+            master_path = os.path.join(config.ENV_DATA.get('cluster_path'), constants.MASTER_IGN)
+            worker_path = os.path.join(config.ENV_DATA.get('cluster_path'), constants.WORKER_IGN)
+
+            self.host = self.helper_node_details['bm_httpd_server']
+            self.user = self.helper_node_details['bm_httpd_server_user']
+            self.private_key = os.path.expanduser(
+                config.DEPLOYMENT['ssh_key_private']
+            )
+
+            self.helper_node_handler = Connection(self.host, self.user, self.private_key)
+            cmd = f"rm -rf {self.helper_node_details['bm_path_to_upload']}"
+            logger.info(self.helper_node_handler.exec_cmd(cmd=cmd))
+            cmd = f"mkdir -m 755 {self.helper_node_details['bm_path_to_upload']}"
```

Comment:
you are calling this command a few times below. How about creating a function for that?

Reply:
If we create func we need to pass paramiko handler

Reply:
we can also do it in a separate PR